modules/tinymce/src/themes/silver/test/ts/browser/editor/buttons/GroupToolbarButtonTest.ts

```diff
@@ -121,4 +122,25 @@ describe('browser.tinymce.themes.silver.editor.buttons.GroupToolbarButtonTest',
       return Promise.resolve();
     })
   );
+
+  it('TINY-9496: onSetup function should run when defining custom group toolbar button', () =>
+    pTestWithEditor({
+      ...defaultToolbarGroupOptions,
+      toolbar: 'test',
+      setup: (editor: Editor) => {
+        editor.ui.registry.addGroupToolbarButton('test', {
+          text: 'test',
+          items: 'alignleft aligncenter alignright',
+          onSetup: (api) => {
+            api.setEnabled(false);
```

Comment:
You could also just have a mutable state to check here since this could in theory be true if the button was disabled by default for what ever reason. So just a `let hasSetupBeenCalled = false;` then locally mutate that and check that in the test assert. That way you would also avoid the long css selector query.

- [ ] Consider using local mutable state

Reply:
Done!